CRM/Contact/BAO/SavedSearch.php

```diff
@@ -360,7 +360,10 @@ public static function create(&$params) {
       if (empty($params['id'])) {
         $params['created_id'] = $loggedInContactID;
       }
-      $params['modified_id'] = $loggedInContactID;
+      // dev/core#2550: Prevent 5.x to 5.36+ upgrade failure
+      if (CRM_Core_BAO_SchemaHandler::checkIfFieldExists('civicrm_saved_search', 'modified_id')) {
```

Comment:
This is the right fix to the problem, but there is IMO a better way to implement it. This provides the same safety check, but without the expense of an extra db query:

```suggestion
      if (array_key_exists('modified_id', self::getSupportedFields())) {
```